homeassistant/components/minecraft_server/__init__.py

```diff
@@ -23,8 +30,20 @@
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Minecraft Server from a config entry."""
 
+    # Check and create API instance.
+    try:
+        api = await hass.async_add_executor_job(
+            MinecraftServer,
+            entry.data.get(CONF_TYPE, MinecraftServerType.JAVA_EDITION),
+            entry.data[CONF_ADDRESS],
+        )
+    except MinecraftServerAddressError as error:
+        raise ConfigEntryError(
+            f"Address in configuration entry is invalid (error: {error}), please remove this device and add it again"
```

Comment:
Please remove this device? That doesn't sound like a solution or suggestion we should offer out of the box. 

When is this needed?

Reply:
Should I remove the suggestion from the log and only keep the error message?

In the previous PR `CONF_HOST` and `CONF_PORT` were replaced by `CONF_ADDRESS`, which is a combination of host and port. For this replacement in the config entry data a migration was added. This raise of `ConfigEntryError` was added for the theoretical possibility of an error caused by the migration. But if you ask me this error should never occur. 

Do you have any other recommendation or should I just edit the log message?

Reply:
I removed the questionable suggestion in the log message. If anything else has to be done/fixed, please unresolve this conversation and let me know what else to do. Thanks 👍